### PR TITLE
Added possible out of shared memory error message

### LINKED ISSUE #5040: Give a better error when we run out of shared memory, instead of "RuntimeError: DataLoader worker (pid 13) is killed by signal: Bus error."
When I set `num_workers=1` or other value greater than 0 in `torch.utils.data.DataLoader`, I get this error.
The detail of the error:
``` 
Traceback (most recent call last):
  File "/opt/project/train.py", line 150, in <module>
    dataset_sizes=dataset_sizes)
  File "/opt/project/train.py", line 51, in train_model
    outputs = model(inputs)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/nn/modules/module.py", line 357, in __call__
    result = self.forward(*input, **kwargs)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/nn/parallel/data_parallel.py", line 64, in forward
    inputs, kwargs = self.scatter(inputs, kwargs, self.device_ids)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/nn/parallel/data_parallel.py", line 75, in scatter
    return scatter_kwargs(inputs, kwargs, device_ids, dim=self.dim)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/nn/parallel/scatter_gather.py", line 30, in scatter_kwargs
    inputs = scatter(inputs, target_gpus, dim) if inputs else []
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/nn/parallel/scatter_gather.py", line 25, in scatter
    return scatter_map(inputs)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/nn/parallel/scatter_gather.py", line 18, in scatter_map
    return list(zip(*map(scatter_map, obj)))
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/nn/parallel/scatter_gather.py", line 15, in scatter_map
    return Scatter.apply(target_gpus, None, dim, obj)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/nn/parallel/_functions.py", line 68, in forward
    outputs = comm.scatter(input, ctx.target_gpus, ctx.chunk_sizes, ctx.dim, streams)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/cuda/comm.py", line 189, in scatter
    outputs.append(chunk.cuda(device, async=True))
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/_utils.py", line 69, in _cuda
    return new_type(self.size()).copy_(self, async)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 172, in handler
    _error_if_any_worker_fails()
RuntimeError: DataLoader worker (pid 13) is killed by signal: Bus error.
```

### PR DESCRIPTION
**Pull request by pbelevich**
Fixes #5040

### CODE REVIEW

File: torch/csrc/DataLoader.cpp [hunk 1]
```diff
@@ -131,6 +131,9 @@ static PyObject *THPModule_errorIfAnyWorkerFails(PyObject *module) {
         std::ostringstream oss;
         oss << "DataLoader worker (pid " << worker_pid << ") is killed "
             << "by signal: " << strsignal(infop.si_status) << ". ";
+        if (infop.si_status == SIGBUS) {
+            oss << "Possibly because out of shared memory. ";
```
Comment (zou3519) [inline torch/csrc/DataLoader.cpp]: nit: we can make this a little more detailed. "It is possible that you are out of shared memory. Please try to raise your shared memory limit."

**Review by zou3519 — APPROVED** [review]

**Review by ssnl — COMMENTED** [review]
There is already a similar message in line 61.

**Review by zou3519 — CHANGES_REQUESTED** [review]
Thanks for pointing that out, @SsnL. The question now is to figure out if it gets triggered when we send a SIGBUS. I was reading the original issue (#5040) and that one implies that the error message at line 61 might not get triggered, but no one has posted a recent copy of it.

**Review by zou3519 — APPROVED** [review]
Based on https://github.com/pytorch/pytorch/issues/5040#issuecomment-531401065, it looks like the signal handlers print before the stack trace. It would be nice for the error message in the signal handler to also appear at the end of the stack trace. 

This PR makes the SIGBUS error message happen at the end of the stack trace, which is great.

**Pull request closed without merging**